lua/lean/infoview.lua

```diff
@@ -27,15 +35,22 @@ function M.update(infoview_bufnr)
       local lines = {}
 
       if result and result.goals then
+        vim.list_extend(lines,
+          {#result.goals == 0 and '▶ goals accomplished 🎉' or
+            #result.goals == 1 and '▶ 1 goal' or
+            string.format('▶ %d goals', #result.goals)})
         for _, each in pairs(result.goals) do
-          vim.list_extend(lines, vim.split(each, '\n', true))
           vim.list_extend(lines, {''})
+          vim.list_extend(lines, vim.split(each, '\n', true))
         end
       end
 
-      for _, diags in pairs(vim.lsp.diagnostic.get_line_diagnostics(current_buffer, cursor[0])) do
-        vim.list_extend(lines, vim.split(diags.message, '\n', true))
-        vim.list_extend(lines, {''})
+      for _, diag in pairs(vim.lsp.diagnostic.get_line_diagnostics(current_buffer, cursor[0])) do
+        local start = diag.range["start"]
+        local end_ = diag.range["end"]
+        vim.list_extend(lines, {'', string.format('▶ %d:%d-%d:%d: %s:',
```

Comment:
I think we need a `+1` for the character too right? Otherwise I see 0 based indices if I have `#check 13` at the beginning of a line.

Reply:
You're right!